src/java/org/apache/cassandra/cql3/constraints/ColumnConstraintScalar.java

```diff
@@ -0,0 +1,167 @@
+        this.relationType = relationType;
+        this.term = term;
+    }
+
+    @Override
+    public void evaluate(Class<? extends AbstractType> valueType, Object columnValue)
+    {
+        Number columnValueNumber;
+        float sizeConstraint;
+
+        try
+        {
+            columnValueNumber = (Number) columnValue;
+            sizeConstraint = Float.parseFloat(term);
+        }
+        catch (NumberFormatException exception)
+        {
+            throw new ConstraintViolationException(param + " and " + term + " need to be numbers.");
+        }
+
+        switch (relationType)
+        {
+            case EQ:
+                if (Float.compare(columnValueNumber.floatValue(), sizeConstraint) != 0)
+                    throw new ConstraintViolationException(param + " value should be exactly " + sizeConstraint);
+                break;
+            case NEQ:
+                if (Double.compare(columnValueNumber.floatValue(), sizeConstraint) == 0)
+                    throw new ConstraintViolationException(param + " value should be different from " + sizeConstraint);
+                break;
+            case GT:
+                if (columnValueNumber.floatValue() <= sizeConstraint)
+                    throw new ConstraintViolationException(param + " value should be larger than " + sizeConstraint);
+                break;
+            case LT:
+                if (columnValueNumber.floatValue() >= sizeConstraint)
+                    throw new ConstraintViolationException(param + " value should be smaller than " + sizeConstraint);
+                break;
+            case GTE:
+                if (columnValueNumber.floatValue() < sizeConstraint)
+                    throw new ConstraintViolationException(param + " value should be larger or equal than " + sizeConstraint);
+                break;
+            case LTE:
+                if (columnValueNumber.floatValue() > sizeConstraint)
+                    throw new ConstraintViolationException(param + " value should be smaller or equal than " + sizeConstraint);
+                break;
+            default:
+                throw new ConstraintViolationException("Invalid relation type: " + relationType);
+        }
+    }
+
+    @Override
+    public void validate(ColumnMetadata columnMetadata) throws InvalidConstraintDefinitionException
+    {
+        if (!(columnMetadata.type instanceof org.apache.cassandra.db.marshal.NumberType))
+            throw new InvalidConstraintDefinitionException(param + " is not a number");
+    }
+
+    @Override
+    public String toString()
+    {
+        return param + " " + relationType + " " + term;
+    }
+
+    @Override
+    public IVersionedSerializer<ColumnConstraint> serializer()
+    {
+        return serializer;
+    }
+
+    @Override
+    public void appendCqlTo(CqlBuilder builder) {
+        builder.append(toString());
+    }
+
+    public static class Serializer implements IVersionedSerializer<ColumnConstraint>
```

Comment:
should not this implement `IVersionedSerializer<ColumnConstraintScalar>`?

Reply:
The reason is that we have a `IVersionedSerializer<ColumnConstraint> serializer();` on the interface.